src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -314,6 +313,85 @@ public void TrackDependency(DependencyTelemetry telemetry)
             this.Track(telemetry);
         }
 
+        //// Preserve original version of 'Track' with TelemetryInitializer being called after iKey check for Devices SDK till we completely move to a different SDK.
+#if NET40 || NET45
```

Comment:
it is hard to understand what's the difference between those two versions of ```Track```. Can you please switch to ``if``` inside the Track method itself?